streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -584,21 +570,24 @@ public StreamThread(final InternalTopologyBuilder builder,
         this.time = time;
         this.streamsMetadataState = streamsMetadataState;
         this.taskManager = taskManager;
-        this.logPrefix = logPrefix(threadClientId);
+        this.logPrefix = String.format("stream-thread [%s] ", threadClientId);
         this.streamsMetrics = streamsMetrics;
         this.restoreConsumer = restoreConsumer;
         this.stateDirectory = stateDirectory;
-        this.rebalanceListener = new RebalanceListener(time, taskManager, this, logPrefix);
+        this.rebalanceListener = new RebalanceListener(time, taskManager, this);
         this.config = config;
         this.stateLock = new Object();
         this.standbyRecords = new HashMap<>();
         this.partitionGrouper = config.getConfiguredInstance(StreamsConfig.PARTITION_GROUPER_CLASS_CONFIG, PartitionGrouper.class);
-        log.info("{} Creating consumer client", logPrefix);
+
+        createLogContext(this.logPrefix);
```

Comment:
We should pass in the `LogContext` from `create`

Reply:
Fixed